From d42fe7d12cdc5f14e1bd9fd13f5d56e2689793dd Mon Sep 17 00:00:00 2001
From: "Yuriy M. Kaminskiy" <yumkam@gmail.com>
Date: Sat, 9 Feb 2019 12:55:11 +0300
Subject: [PATCH 1/6] Fix compilation on i386

---
 include/gf_complete.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/gf_complete.h b/include/gf_complete.h
index c4783e8..9436eb8 100644
--- a/include/gf_complete.h
+++ b/include/gf_complete.h
@@ -19,6 +19,10 @@
   #ifdef __SSE4_1__
     #include <smmintrin.h>
   #endif
+  #ifdef __i386
+    #define _mm_insert_epi64(A,B,C) _mm_insert_epi32(_mm_insert_epi32((A),(uint32_t)(B),(C)*2),(uint32_t)((uint64_t)(B)>>32),(C)*2+1)
+    #define _mm_extract_epi64(A,C) ((((uint64_t)_mm_extract_epi32((A),(C)*2+1))<<32)|(uint32_t)_mm_extract_epi32((A),(C)*2))
+  #endif
 #endif
 
 #ifdef INTEL_SSSE3
-- 
2.11.0

