From 03ea8244d34a2800747bac0f0a4355a017f31a7b Mon Sep 17 00:00:00 2001
From: "Yuriy M. Kaminskiy" <yumkam@gmail.com>
Date: Sun, 10 Feb 2019 11:54:23 +0300
Subject: [PATCH 3/6] Enable runtime cpudetection on i386

---
 src/gf_cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gf_cpu.c b/src/gf_cpu.c
index f65131f..e113051 100644
--- a/src/gf_cpu.c
+++ b/src/gf_cpu.c
@@ -20,7 +20,7 @@ int gf_cpu_supports_intel_sse3 = 0;
 int gf_cpu_supports_intel_sse2 = 0;
 int gf_cpu_supports_arm_neon = 0;
 
-#if defined(__x86_64__)
+#if defined(__x86_64__) || defined(__i386)
 
 /* CPUID Feature Bits */
 
-- 
2.11.0

